src/metabase/actions/execution.clj

```diff
@@ -203,22 +203,41 @@
       (execute-custom-action! action request-parameters)
       (throw (ex-info (tru "Unknown action type {0}." (name (:type action))) action)))))
 
+(defn- execute-table-action!
+  [kind table-id request-parameters]
+  (let [args
+        {:type     :query
+         :query    {:source-table table-id}
+         :database (api/check-404 (t2/select-one-fn :db_id [:model/Table :db_id] table-id))
```

Comment:
And I think this one is optional, so worth trying to remove also.

Reply:
PS it can't 404 because of our FK constraint

Reply:
c7112a6e0271